tests/flow.ml

```diff
@@ -0,0 +1,219 @@
+module Unix_scheduler = struct
+  type +'a t = 'a
+
+  let bind x f = f x
+  let return x = x
+end
+
+module Conduit = Conduit.Make(Unix_scheduler)(Bytes)(String)
+
+let recv =
+  let pp ppf = function
+    | `Input len -> Fmt.pf ppf "@[<1>(`Input@ %d)@]" len
+    | `End_of_flow -> Fmt.string ppf "`End_of_flow" in
+  let equal a b = match a, b with
+    | `Input a, `Input b -> a = b
+    | `End_of_flow, `End_of_flow -> true
+    | _ -> false in
+  Alcotest.testable pp equal
+
+let send = Alcotest.int
+
+let error =
+  let pp ppf = function
+    | #Rresult.R.msg as v -> Rresult.R.pp_msg ppf v
+    | `Not_found -> Fmt.string ppf "`Not_found" in
+  let equal a b = match a, b with
+    | `Msg a, `Msg b -> a = b
+    | `Not_found, `Not_found -> true
+    | _ -> false in
+  Alcotest.testable pp equal
+
+module Memory_flow0 = struct
```

Comment:
Can you add this in `Conduit` directly (the usual name for this is `Loopback`) ?

Reply:
I'm not sure because:
- such "protocol" works with `string` as an input and `bytes` as an output - so it requires an application of `Conduit.Make` which differs from the used application with `cstruct` by `conduit-lwt` and `conduit-async` (see l. 8). At the end, it will not be possible to compose it with others `conduit-*`.

Note that I tried to abstract such types into witnesses but it reveals something more complex then.

- for me, such impl. is only relevant about tests. In this view, we should not expose it.

Reply:
Well, having an efficient in-memory flow to connect various process is very useful in some situation (not always for tests) ; it would be a shame if every user would have to write one. 

I agree that maybe Memory_flow0 or Memory_flow1 are not the best candidates for these as they do not work on cstructs as the rest of the client library.

Reply:
I see your point, I can try to provide something with `Cstruct.t` as `conduit-tls` does.

Reply:
I try to figure out a bit about this part and I think it should be a part of `mirage-flow` instead to integrate it in `ocaml-conduit`. One crucial point is: `conduit` uses an other interface than `mirage-flow`.

A plan can be:
1) release `ocaml-conduit` with the conflict between `mirage-flow` and `Conduit.FLOW`
2) start to deprecated the `mirage-flow` interface (as mentioned into `conduit-lwt.flow`)
3) update `mirage-flow` interface and do a revdeps on it (specially, update `cohttp`)
4) implement `Loopback` into `mirage-flow` then

Reply:
About [deprecation of `mirage-flow`](https://github.com/mirage/ocaml-conduit/pull/311/files#diff-38af78a963122b1fbe844f31d53444e9R1).